frontend/src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -11,6 +13,7 @@ export interface CheckboxGroupOption {
 interface ComponentProps {
   options: CheckboxGroupOption[];
   selectedOptions?: string[];
+  usage?: 'select' | 'groupby' | 'join';
```

Comment:
Nope ... CheckboxGroup doesn't know about any of this query builder stuff ... in fact, it can be used to select anything that can be rendered by checkboxes.
If what you wanted was a way to toggle the `selectAll` check, I suggest a simple `selectAll` boolean prop that defaults to false ... I recall this was mentioned before.

Reply:
Renamed and updated to a boolean